api/src/main/java/org/apache/iceberg/io/FileIOMetrics.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.io;
+
+import java.io.Serializable;
+import java.util.Map;
+
+/**
+ * Optional interface for FileIO implementations to report metrics related
+ * to read/write operations.
+ */
+public interface FileIOMetrics extends Serializable {
+  String READ_BYTES = "read.bytes";
+  String READ_OPERATIONS = "read.operations";
+  String WRITE_BYTES = "write.bytes";
+  String WRITE_OPERATIONS = "write.operations";
+
+  default void initialize(Map<String,String> properties) {}
+
+  interface Counter<T> {
+    void increment(T amount);
```

Comment:
Maybe add `default void increment() { increment(1); }`?

Reply:
I've incorporated this, which reads better for single increments.